src/bin/ion/commands/beta/generate/context.rs

```diff
@@ -0,0 +1,60 @@
+use crate::commands::beta::generate::utils::Language;
+use serde::Serialize;
+use tera::Tera;
+
+/// Represents a context that will be used for code generation
+pub struct CodeGenContext {
+    // Represents the templating engine - tera
+    pub(crate) tera: Tera,
+    pub(crate) language: Language,
+    // Initially the data_model field is set to None.
+    // Once an ISL type definition is mapped to a data model this will have Some value.
+    pub(crate) data_model: Option<DataModel>,
+    // Represents a counter for naming anonymous type definitions
+    pub(crate) anonymous_type_counter: usize,
+}
+
+impl CodeGenContext {
+    pub fn new(language: Language) -> Self {
+        Self {
+            language,
+            data_model: None,
+            anonymous_type_counter: 0,
+            tera: Tera::new("src/bin/ion/commands/beta/generate/templates/**/*.templ").unwrap(),
+        }
+    }
+
+    pub fn with_data_model(&mut self, data_model: DataModel) {
+        self.data_model = Some(data_model);
+    }
+
+    pub fn with_initial_data_model(&mut self) {
+        // Initially the data model is set to None, this will be set with Some(_) value when data model is determined in code generation process
+        self.data_model = None;
+    }
+
+    /// Returns a string that represent the template name based on data model type.
+    pub fn template_name(&self) -> &str {
+        if let Some(data_model) = &self.data_model {
+            return match (&self.language, data_model) {
+                (
+                    Language::Rust,
+                    DataModel::Struct | DataModel::UnitStruct | DataModel::SequenceStruct,
+                ) => "struct",
+                (
+                    Language::Java,
+                    DataModel::Struct | DataModel::UnitStruct | DataModel::SequenceStruct,
+                ) => "class",
+            };
+        }
+        "" // Default value is an empty string
+    }
+}
+
+/// Represents a data model type that can be used to determine which templates can be used for code generation.
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub enum DataModel {
+    UnitStruct,     // a struct with a scalar value (used for `type` constraint)
+    SequenceStruct, // a struct with a sequence value (used for `element` constraint)
+    Struct,
+}
```

Comment:
I think some of the PR tour comments became outdated due to a reorganization of the code.
https://github.com/amazon-ion/ion-cli/pull/61#discussion_r1391622634
element -> SequenceStruct(creates a struct with single field that stores the sequence as a Vec)
e.g. 
```rust 
struct Foo { value: Vec<String> }
```
fields -> Struct (creates a struct with given field name and value/data type pairs)
e.g. 
```rust 
struct Foo { name: String, id: i64 }
```
type - >UnitStruct (creates a struct with single field that has the data type provided with type constraint)
e.g. 
```rust 
struct Foo { value: String }
```

Reply:
> I think some of the PR tour comments became outdated due to a reorganization of the code.

Ugh, my PR tour comments get bit by this _all the time_. I usually notice problems in my own code as I'm writing up the tour, so I go to fix it and GitHub cheerfully throws my comments out the window. -_-

Thank you for explaining. I think this:

> type -> UnitStruct (creates a struct with single field that has the data type provided with type constraint)

is a bit misleading since (in Rust) a unit struct is a struct with zero fields (e.g. `struct Foo;` or `struct Foo {}`) and here it's a struct with one field.

What do you think of this?:
```rust
pub enum DataModel {
    Value,
    Sequence,
    Struct
}
```
?

It's a bit more abstract--notice that it doesn't mention the target kind at all--which seems appropriate for a data model.

Reply:
Yes, I think that makes sense. Another option would be to use something similar to serde where it uses `newtype_struct`. But overall there will be more data models added as we support more constraints so I am fine to go with Value, Sequence, .. now and if required we can change variant names later as we add more data model since it is not going to impact the generated code anyway.